buildenv/jenkins/jobs/infrastructure/artifactory_cleanup.groovy

```diff
@@ -0,0 +1,131 @@
+* 
+* Parameters: 
+*   JOB_TYPE: Choice
+*             Expected values: one of the following
+*             Time,                // This tells the script to delete all artifacts that are older that were created over x amount of days ago
+*             Number of Builds     // This tells the script to keep x amount of artifacts from a specific project
+*   JOB_TO_CHECK: String – The folder to keep x amount of builds in 
+*   ARTIFACTORY_SERVER: String - The artifactory server to clean up
+*   ARTIFACTORY_REPO: String - The artifactory repo to clean up
+*   ARTIFACTORY_NUM_ARTIFACTS: String - How many artifacts to keep in artifactory
+*   ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS: String - How many days the artifacts can stay in artifactory (if any are exceeded, they are deleted)
+**/
+
+timestamps {
+    node('worker') {
+        checkout scm
+        def variableFile = load 'buildenv/jenkins/common/variables-functions'
+        variableFile.parse_variables_file()
+        variableFile.set_artifactory_config()
+
+        def ARTIFACTORY_SERVER = sever.getUrl()
+        def ARTIFACTORY_REPO = params.ARTIFACTORY_REPO ? params.ARTIFACTORY_REPO : env.ARTIFACTORY_REPO
+        def JENKINS_DOMAIN = params.JENKINS_DOMAIN ? params.JENKINS_DOMAIN : ''
+        def ARTIFACTORY_NUM_ARTIFACTS = params.ARTIFACTORY_NUM_ARTIFACTS ? params.ARTIFACTORY_NUM_ARTIFACTS : env.ARTIFACTORY_NUM_ARTIFACTS
+        def ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS = params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS ? params.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS : env.ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS
+
+        if (params.JOB_TYPE == 'Time'){
+            cleanupTime(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, ARTIFACTORY_DAYS_TO_KEEP_ARTIFACTS)
+        } else if (params.JOB_TYPE == 'Number of Builds'){
+            if (params.JOB_TO_CHECK){
+                cleanupBuilds(ARTIFACTORY_SERVER, ARTIFACTORY_REPO, JENKINS_DOMAIN, params.JOB_TO_CHECK, ARTIFACTORY_NUM_ARTIFACTS)
+            } else {
+                error 'Please input a job to cleanup'
+            }
+        }
+    }
+}
+
+def cleanupBuilds(artifactory_server, artifactory_repo, jenkins_domain, jobToCheck, artifactory_num_artifacts){
+    // This parameter is originally a string and needs to be casted as an Integer
+    def artifactory_num_artifacts = artifactory_num_artifacts as Integer 
+
+    stage('Discover Stored Artifacts'){
+        echo "Cleaning up ${jobToCheck}"
+        echo "Keeping the latest ${artifactory_num_artifacts} builds"
+
+        def request = httpRequest authentication: env.ARTIFACTORY_CREDS, consoleLogResponseBody: true, url: "${artifactory_server}/api/storage/${artifactory_repo}/${jenkins_domain}/${jobToCheck}"
+
+        data = readJSON text: request.getContent()
+        numberOfArtifacts = data.children.size()
+        echo "There are ${numberOfArtifacts} builds"
+    } 
+    stage('Delete Old Artifacts'){
+        if (numberOfArtifacts > artifactory_num_artifacts){
```

Comment:
Can we change `artifactory_num_artifacts` to something with either `max` or `keep` so this is a little easier to understand?